File: Java-version/LeetCode/src/com/jooc/GazeOffer/Solution_22.java
package com.jooc.GazeOffer;

import com.jooc.components.ListNode;

public class Solution_22  {
    public ListNode getKthFromEnd(ListNode head, int k) {
        ListNode before = head;
        ListNode later = head;

        for(int i = 0; i < k; i++){
            later = later.next;
        }

        while(later != null){
            before = before.next;
            later = later.next;
        }

        return before;
    }
}

File: Swift-version/LeetCode/LeetCode/Solutions/GazeOffer/Solution_22.swift
//
//  Solution_22.swift
//  LeetCode
//
//  Created by 齐旭晨 on 2021/3/30.
//

import Foundation

class Solution_22: Solution{
    func main() {

    }
    
    func getKthFromEnd(_ head: ListNode?, _ k: Int) -> ListNode? {
        var slow = head
        var fast = head
        
        for _ in 0..<k{
            fast = fast!.next
        }
        
        while fast != nil{
            slow = slow?.next
            fast = fast?.next
        }
        return slow
    }
}
